subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -22,8 +22,62 @@ Gradle build scripts combine the qualities of declarative build logic, expressiv
 The third-party link:https://github.com/nebula-plugins/gradle-lint-plugin[Gradle lint plugin] helps with enforcing a desired code style in a build script if you are looking for appropriate linting automation.
 ====
 
-[[sec:declare_task_group_and_description]]
-=== Declare the group and description for tasks
+=== Avoiding Gradle internal APIs
+
+Use of Gradle internal APIs in plugins and build scripts causes builds to break when either Gradle or plugins change.
```

Comment:
I'd rather say "has the potential to break". Sometimes we change but it's not a change in the signature of a method.